Hash_map/include/CGAL/Handle_hash_function.h

```diff
@@ -33,10 +33,20 @@ namespace internal{
       std::size_t
       operator()(const H& h)
       {
-        return std::size_t(&*h) /
+        return std::size_t(h.operator->()) /
```

Comment:
If you'd like to avoid the functional-style cast here (which is equivalent to the overpowered C-style cast), you could use the appropriate `reinterpret_cast<std::size_t>` instead, which is idiomatic for reinterpreting pointers as integers and vice versa.

Reply:
Maybe this can be applied more generally using clang-tidy: https://clang.llvm.org/extra/clang-tidy/checks/google-readability-casting.html

Reply:
I am in favor of using that clang rule. @maxGimeno Could you give it a try? First on a small set of CGAL, like a few CGAL plugins.